### DISCUSSION TITLE
Inverter-SmartCharger-Dongle

### DISCUSSION THREAD

**Discussion by Criaga** [wlcrs/huawei_solar#750] [Connecting to the inverter]
Inverter Sun2000 6ktl - fw:141
Battery Luna 
Dongle 
Smart Charger Huawei 

Good morning everyone. How did you integrate the inverter with Home Assistant? 
In the last few weeks, using the Huawei  Solar integration, I have had constant disconnections. 
I use the IP that the dongle has on my network.

If I disconnect the smart charger I can use the integration. You think it's a Modbus problem. I also activated Modbus Proxy with IP dongle and port 502 but it doesn't work

Help

**Comment by Roving-Ronin** [comment]
@Criaga See: https://github.com/wlcrs/huawei_solar/issues/491

Short version is that if the Charger is connecting to the sDongle in the Inverter, then it is using the single modbus connection that the sDongle supports, hence the HA/WLCRS will not be able to connect.

Suggest you might be able to look into the DDSU / DTSU-666-FE  meter, that supports connecting to the Charger via an ethernet connection. Off the top of my head, I've forgotten if you could install this newer meter and have it connected to the Inverter (via RS485... not sure re IP connection) and then have the Charger connected via the Ethernet connection (cross over cable arrangement).

Have a look at the above link, most of the documentation should be there, showing the 'Network' options (as Huawei refers to RS485 / Ethernet etc).

**Reply by Criaga** [reply]
I tried to install version 1.3.3 and as of this morning it has no disconnects. Now I'm connected via dongle IP /port 502/slave 1. Let's see what happens in the next few hours, but it seems very stable

**Reply by Roving-Ronin** [reply]
@Criaga 
FYI WLCRS v1.4 has been released. Worth the update !!!!!
BTW you don't say if you inverter is an L0 / L1 / M0 / M1 etc (1p or 3p).

**Reply by Criaga** [reply]
6ktl-L1 Single Phase Inverter

I had updated to version 1.4 and then to version 1.4.1. That's when the problems started

**Reply by Criaga** [reply]
@Roving-Ronin 

Sorry for the inconvenience! Since this morning the integration works perfectly without any disconnection... moreover the Huawei smart charger works perfectly without errors.

As I was saying, I installed version 1.3.3 and connected via the IP of the dongle with port 502. I don't know why with version 1.4.1 it continually lost the connection.

**Reply by Roving-Ronin** [reply]
Does anything appear in HA for the charger (I'd assume no)?

V1.4 shouldn't make any difference in regards the connection, but either way the change in line with updates to HA, it should be more efficient in polling. Had a couple of sites with 1.3.3 --> 1.4.0Alpha --> v1.4 and no real issues, the only one atm is really about the 'Remaining Charge/Discharge Time' staying at 0min, on the 'Batteries' (that shows the summed up view for all installed batteries).

**Reply by Criaga** [reply]
Nothing appears on HA about the wallbox.

Honestly, I'm no expert at all, but the mystery remains. With version 1.3.3, since yesterday morning, no disconnection.
 I was thinking of using Modbus Proxy but for now I don't do anything 😅

**Reply by Roving-Ronin** [reply]
What version of firmware are you running on your:

Inverter 5KTL-L1  :  v ?
LUNA ESS            :  v ?
sDongle               :  v ?
Charger               :  v?

From what I've seen, drop outs are either an issue re the firmware your using (and the compatability between the versions and the other devices versions) or are simply WiFi issues on the home network setup.

**Reply by Criaga** [reply]
Inverter : V200R001C00SPC141
Luna : V100R002C00SPC621
SDongle : V200R022C10SPC114
Charger : V100R023C10SPC020

**Reply by Roving-Ronin** [reply]
Jesus... how's the wind in your hair, at the bleeding edge of firmware versions.   ;-) LoL.

They are basically the latest releases... actually, have you also updated the modbus (comes up as 'MBUS' in the choose of things to update in FusionSolar Portal) also ? Latest is is  SmartMBUSV100R001C00SPC335 (depending upon which country you are, I'm finding some Huawei local support people device what is the latest firmware they'll "allow" you to see/choose from, so you might see SPC330 as the latest available).

**Reply by Criaga** [reply]
Call me FLASH 😂

MBus: V100R001C00SPC335

**Reply by Criaga** [reply]
I found this on the Huawei site. As an ignorant person, is it possible that the communication between smart charger and inverter is on two different ports compared to HA?


![IMG_4030](https://github.com/wlcrs/huawei_solar/assets/133338962/103d8a2f-a988-4265-b1a5-87a2c958cd8a)

**Reply by Roving-Ronin** [reply]
You might want to add your configuration into here, https://github.com/wlcrs/huawei_solar/wiki/Confirmed-firmware-versions  but add a comment at the end re intermittent connection drop outs. PS. Its sorted by type of inverter, then size of inverter and then within that by the oldest to newest inverter firmware version.  i.e. add yours to the bottom of the L1 5kW list under the SPC138 version.

The sDongle is on tcp/502 (usually... 6607 is the other option), the Charger either connects to tcp/502 or it fails to connect at all. The only other number that changes is the modbus ID#  (that you put into the last line of the WLCRS install in HA.. i.e. 1,2 or 2,1 in my case).

As mentioned before, in the current Huawei user manuals with a site having a DDSU-666H meter that is RS485 connected to the inverter and the inverter has an sDongle for LAN/WAN connectivity. It shows the EV Charger connecting over the LAN (IP connection) to the sDongle, whereby it uses that as a 'Virtual Meter' to see the power flows to/from the grid that the DDSU-666H is providing to the inverter.
Given the sDongle can only accept a SINGLE modbus connection this would cause it (or HA) to fail to get a connection. I have no idea if the EV Charger permamently maintains a connection to the sDongle, or if when not being used if it hybernates, that may then see the modbus connection be dropped and the HA/WLCRS reconnect again... ?

As per this post: https://github.com/wlcrs/huawei_solar/issues/491#issuecomment-2094635359

Short of having EMMA's (not supported in WLCRS currently/can't get documentation to add support still) and connecting the EV Charger to it via the LAN, the other option I remember is as per the diagram in that thread/post, that shows having to install an additional meter, this being the new DDSU-666FE (the diagram shows DSTU..thats just the prefix for the 3phase version), next to your existing DDSU-666H (RS485) meter, with the new meter being used to provide data solely to the EV Charger.

![image](https://github.com/wlcrs/huawei_solar/assets/108674933/ca2ded9a-eb2a-4512-ab6f-23963de426a1)
